src/js/AppStore.js

```diff
@@ -0,0 +1,196 @@
+import React from 'react';
+import Modal from 'react-modal'
+import { connect } from 'react-redux';
+import store from './store';
+import { flags } from './Flags'
+
+import AppHeader from './containers/Header';
+import HScrollMenu from './HScrollMenu';
+import AppServices from './containers/AppServices';
+import ConfirmAlert from './ConfirmAlert';
+
+import BCController from './Controllers/BCController';
+import fileSystemController from './Controllers/FileSystemController'
+
+import { updateAvailableAppStoreApps, appStoreAddAppPendingSetAppProperties } from './actions';
+
+class AppStore extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            confirmID: null,
+            confirmApp: null
+        }
+
+        this.cancel = this.cancel.bind(this);
+        this.confirm = this.confirm.bind(this);
+        this.onSelection = this.onSelection.bind(this);
+    }
+
+    cancel() {
+        this.setState((state, props) => {
+            return {
+                confirmID: null,
+                confirmApp: null
+            }
+        });
+    }
+
+    confirm() {
+        fileSystemController.subscribeToEvent('InstallApp', (success, params) => {
+            if (!success || !params.appUrl) {
+                console.error('error encountered when installing app');
+            }
+
+            fetch(params.appUrl + 'manifest.js')
```

Comment:
Might want to add a "/" before manifest in case the appURL does not include that as the last character.

Reply:
We decided it is mandatory for the `appUrl` to end in a `/`

Reply:
Is that documented somewhere?